domain/service/servicedetails_store.go

```diff
@@ -73,6 +62,32 @@ func (s *storeImpl) GetAllServiceDetails(ctx datastore.Context, since time.Durat
 		}
 
 		parentMap[d.ParentServiceID] = struct{}{}
+
+		if r != nil && !r.MatchString(d.Name) {
+			continue
+		}
+
+		if query.Since > 0 && d.UpdatedAt.Before(since) {
+			continue
+		}
+
+		if query.Tenants && d.ParentServiceID != "" {
```

Comment:
Makes sense for regex. However the "since" parameter can filter out a lot if it is being used by the client.  I'm thinking of issues with very large installs. So params that can be used should be used to filter at the elastic level.